spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -71,6 +71,8 @@ statement
     | ALTER TABLE multipartIdentifier DROP PARTITION FIELD transform                        #dropPartitionField
     | ALTER TABLE multipartIdentifier REPLACE PARTITION FIELD transform WITH transform (AS name=identifier)? #replacePartitionField
     | ALTER TABLE multipartIdentifier WRITE writeSpec                                       #setWriteDistributionAndOrdering
+    | ALTER TABLE table=multipartIdentifier SET IDENTIFIER_KW FIELDS '(' fields+=multipartIdentifier (',' fields+=multipartIdentifier)* ')' #setIdentifierFields
```

Comment:
You might consider allowing more flexible syntax here, like `SET IDENTIFIER FIELD user_id` since `FIELD` will make more sense to users. Should the parens be optional?

Could you move the `fields` to a `fieldList` rule so that we don't have two copies of it?

Reply:
I think we have the following options to choose from:
- ADD IDENTIFIER FIELD
- ADD IDENTIFIER FIELDS
- SET IDENTIFIER FIELD
- SET IDENTIFIER FIELDS
- DROP IDENTIFIER FIELD
- DROP IDENTIFIER FIELDS

I am choosing to implement the batch operations because that allows one single commit to add or drop multiple fields. Personally I don't feel it is hard for users to specify `SET IDENTIFIER FIELDS (user_id)`, given that all Hive DDLs are in this batch format. Do you think it is worth having a clause also for single field modifications in this case?

If we want to have a way to only add new fields without the need to know current fields, I can also add a `ADD IDENTIFIER FIELDS`.

Reply:
My comment was based on similar reasoning for other DDL. For example, it is better to support both `ADD COLUMNS` and `ADD COLUMN` interchangeably so that users don't get needless syntax errors when they use a plural or singular term with the "wrong" number of columns.

I think that the problem extending that to this is that it isn't clear that `SET IDENTIFIER FIELD x` actually replaces all existing identifier fields, which I think is why you're mentioning not wanting to use the `ADD IDENTIFIER FIELD` syntax. I think that if there is confusion now, then there would be more confusion for users, so maybe it is better to only support `SET IDENTIFIER FIELDS (...)`. Though I would make the parens optional.

> . . . given that all Hive DDLs are in this batch format

I don't think that we want to take inspiration from Hive very much. Better to use Postgres or Trino as reference points.

> If we want to have a way to only add new fields without the need to know current fields, I can also add a ADD IDENTIFIER FIELDS.

This wasn't my intent and I would not want to have confusion, so let's not allow using `FIELD` interchangeably.

Reply:
> I would make the parens optional

Do you mean to also not have parens optional when there are multiple fields, like:

```
SET IDENTIFIER FIELDS first_name, last_name
```

Or is it only for the case when there is 1 field?

Reply:
Yeah, I would make it possible to use that syntax. No need for users to remember that they need parens when we don't actually need them to parse correctly.

Reply:
Okay that makes sense. I will update with all the other comments, thanks!